### DISCUSSION TITLE
Drupal Module Advanced Varnish .vcl

### DISCUSSION THREAD

**Discussion by friedinando** [ddev/ddev#6148] [Q&A]
Is there a guide on how the default.vcl should be modified to use the Drupal Advanced Varnish Module (AVM) with DDEV?

They recommend using the following default.vcl:
https://git.drupalcode.org/project/adv_varnish/-/blob/4.x/varnish/default.vcl

How should the AVM default.vcl be merged with the DDEV default.vcl?

Thanks.

**Comment by rfay** [comment]
DDEV does not have varnish by default. I imagine you're interested in using the [ddev-varnish add-on](https://github.com/ddev/ddev-varnish). 

As it says there, you may want to edit the default.vcl, removing the `#ddev-generated` that exists. Maybe you'll find other techniques. If vcl's have a #include functionality, it could include an external file.

**Reply by friedinando** [reply]
Hello rfay! Thank you for the answer.

Yes, I have the add-on already installed, with the #ddev-generated code removed. The add-on includes this [default.vcl](https://github.com/ddev/ddev-varnish/blob/main/varnish/default.vcl).

What I need to do is to "merge" the default.vcl from the DDEV add-on with the default.vcl from the advanced Varnish module, but I don't know the correct way to do this without affecting the DDEV functionality.

Should I replace all the content within the sub vcl_recv and sub vcl_synth sections from the DDEV add-on, with the default.vcl from the advanced Varnish module? Or is there a specific approach I should follow to ensure that the DDEV functionality is not affected?

**Reply by penyaskito** [reply]
It's been some years since I touched a vcl file, but what I would do:

1. Copy the varnish_adv vcl into the add-on one.
2. Edit backend default with host and port from ddev add-on one.
3. Add to the very beginning of vcl_recv the vcl_recv code from ddev add-on
4. Add to the very beginning of vcl_synth the vcl_synth code from ddev add-on

As far as I can see, that will make mailpit to work, which is the only service affected from DDEV.
